packages/firebase_ml/lib/src/firebase_model_download_conditions.dart

```diff
@@ -0,0 +1,56 @@
+part of firebase_ml;
+
+/// Conditions to download remote models.
+class FirebaseModelDownloadConditions {
+  final bool _requiredWifi;
+  final bool _requiredDeviceIdle;
+  final bool _requiredCharging;
+
+  FirebaseModelDownloadConditions._builder(FirebaseModelDownloadConditionsBuilder builder) :
```

Comment:
I don't think you really need this builder class because Dart > Java , just make them 3 named params that defaults to `false`

And the getters can be removed too
https://dart.dev/guides/language/effective-dart/usage#dont-wrap-a-field-in-a-getter-and-setter-unnecessarily

Reply:
Adrian and I thought it would be better to use builders to stay consistent with java structure. I also saw builders implemented in dart style guide  - https://dart.dev/guides/language/language-tour. 
The reason for getters - is because I want these values to be read-only. 

I can achieve the same by making them final and without a builder. Should I do that instead?

Reply:
After discussion with Elvis removed builders in favour of Dart syntactic sugar.